### PR TITLE
Upgrade to go1.19.2

### PR DESCRIPTION
**Pull request by analogue**
Context: [NET-240](https://hashicorp.atlassian.net/browse/NET-240)

Changes proposed in this PR:
- Upgrade to go 1.19.2
- Upgrade to gotestsum 1.8.2

How I've tested this PR:
- unit and acceptance tests are ✅  

How I expect reviewers to test this PR:
👀 

Checklist:
- [ ] Tests added
- [x] CHANGELOG entry added 
  > HashiCorp engineers only, community PRs should not add a changelog entry.
  > Entries should use present tense (e.g. Add support for...)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -614,7 +616,7 @@ jobs:
 
   cleanup-azure-resources:
     docker:
-      - image: docker.mirror.hashicorp.services/hashicorpdev/consul-helm-test:0.14.0
+      - image: docker.mirror.hashicorp.services/hashicorpdev/consul-helm-test:0.15.0
```
[the remark this answers is not in the document]
Comment (analogue) [reply .circleci/config.yml:619]: Nice catch!

**Review by jmurret — CHANGES_REQUESTED** [review]
can you add a changelog entry and please leave that reminder from the PR template in your PR?  The changelog should be similar to this: https://github.com/hashicorp/consul-k8s/pull/1292/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR2-R7

**Comment by jmurret** [comment]
also, this is missing the following files, which means you need to run `go mod tidy` in each of these directories after you modify `go.mod`.
- acceptance/go.sum
- cli/go.sum
- control-plane/go.sum

For reference, I am auditing your PR against the two PRs linked in the JIRA ticket.

File: .circleci/config.yml [hunk 2]
```diff
@@ -22,21 +22,23 @@ gke-terraform-path: &gke-terraform-path charts/consul/test/terraform/gke
...
-            rm go1.18.3.linux-amd64.tar.gz
+            wget https://golang.org/dl/go1.19.2.linux-amd64.tar.gz
+            sudo rm -rf /usr/local/go && sudo tar -C /usr/local -xzf go1.19.2.linux-amd64.tar.gz
+            rm go1.19.2.linux-amd64.tar.gz
             echo 'export PATH=$PATH:/usr/local/go/bin' >> $BASH_ENV
 
-            wget https://github.com/gotestyourself/gotestsum/releases/download/v1.6.4/gotestsum_1.6.4_linux_amd64.tar.gz
-            sudo tar -C /usr/local/bin -xzf gotestsum_1.6.4_linux_amd64.tar.gz
-            rm gotestsum_1.6.4_linux_amd64.tar.gz
+            wget https://github.com/gotestyourself/gotestsum/releases/download/v1.8.2/gotestsum_1.8.2_linux_amd64.tar.gz
```
Comment (jmurret) [inline .circleci/config.yml:39]: 👏 🎉 🙏 nice!

**Comment by analogue** [comment]
> also, this is missing the following files, which means you need to run `go mod tidy` in each of these directories after you modify `go.mod`.
> 
>     * acceptance/go.sum
> 
>     * cli/go.sum
> 
>     * control-plane/go.sum
> 
> 
> For reference, I am auditing your PR against the two PRs linked in the JIRA ticket. I think this and the changelog entry are the missing items.

I re-ran `go mod tidy` on those files, but it didn't yield any changes.

**Review by jmurret — APPROVED** [review]
👍 ❤️ :shipit:

**Review by t-eckert — APPROVED** [review]
Great work!

**Pull request merged**